HHS/OPRE-OPS
# Ops 448 jwt auth z

## What changed

- Flushed out the creation of Access and Refresh Tokens from the `backend`
- Updated the `frontend` to support storing the Access token, and provide them with each request to the `backend`
- Updated `BaseView` to require valid tokens `jwt_required()`, which propagates up to all endpoints
- Added unit tests to test for valid tokens on endpoint requests, and ensure they return 401 when not present
- Created a `auth_client` for use in pytest; which facilitates providing a pre-authenticated client (Authorization header populated)
- Wired up the unit tests to hit the PostgreSQL DB instead of the fallback SQLite
- Updated all unit tests / e2e tests to reflect the new data from our dev data
- Created a basic (developer) `frontend` page for `UserDetails` - mostly for testing
- Updated the Login button header, to include a link for the logged in user.

## Issue

#443 
#448
#656 

## How to test

`PyTest`
Manual - Login should now work, and accessing an `Portfolios` or `CANs` shouldn't return results unless you're logged in.

## Screenshots

![image](https://user-images.githubusercontent.com/56687505/217628094-0869b5f5-41de-4ce9-8520-4b601123398a.png)
